test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -77,9 +126,111 @@ expectResponseCode
 expectResponseCode want (got, _) =
     got `shouldBe` want
 
+expectFieldEqual
+    :: (MonadIO m, MonadFail m, Show a, Eq a)
+    => Lens' s a
+    -> a
+    -> (HTTP.Status, Either RequestException s)
+    -> m ()
+expectFieldEqual getter a (_, res) = case res of
+    Left e  -> wantedSuccessButError e
+    Right s -> (view getter s) `shouldBe` a
+
+--
+-- Lenses
+--
+addressPoolGap :: HasType (ApiT AddressPoolGap) s => Lens' s Word8
+addressPoolGap =
+    lens _get _set
+  where
+    _get :: HasType (ApiT AddressPoolGap) s => s -> Word8
+    _get = getAddressPoolGap . getApiT . view typed
+    _set :: HasType (ApiT AddressPoolGap) s => (s, Word8) -> s
+    _set (s, v) = set typed (ApiT $ unsafeMkAddressPoolGap v) s
+
+balance :: HasType (ApiT WalletBalance) s => Lens' s WalletBalance
```

Comment:
Not sure if we need this one after all, if we're already checking both the available balance and the total balance independently?

Reply:
Yeah, checking independently `balanceAvailable` and `balanceTotal` serves for better readability in the scenario than `expectFieldEqual balance (WalletBalance (Quantity 0) (Quantity 0)) r`. Will remove balance...